client/wp-admin-scripts/onboarding-homepage-notice/index.js

```diff
@@ -2,73 +2,103 @@
 /**
  * External dependencies
  */
-import { select, subscribe, dispatch } from '@wordpress/data';
+import { dispatch } from '@wordpress/data';
 import { __ } from '@wordpress/i18n';
 import apiFetch from '@wordpress/api-fetch';
+import domReady from '@wordpress/dom-ready';
 
 /**
  * WooCommerce dependencies
  */
 import { getAdminLink } from '@woocommerce/navigation';
 
 /**
- * Displays a notice on page save and updates the hompage options.
+ * Returns a promise and resolves when the post begins to publish.
  *
- * Adapted from WSUWP Help Docs by Washington State University (GPL v2).
- * https://github.com/washingtonstateuniversity/WSUWP-Plugin-Help-Docs/blob/master/src/_js/blocks/notices.js
+ * @return {Promise} Promise for overlay existence.
  */
-const onboardingHomepageNotice = () => {
-	const post = select( 'core/editor' ).getCurrentPost();
+const saveStarted = () => {
+	if ( document.querySelector( '.editor-post-publish-button' ) === null ) {
+		const promise = new Promise( resolve => {
+			requestAnimationFrame( resolve );
+		} );
+		return promise.then( () => saveStarted() );
+	}
 
-	const noticeMeta = {
-		wasPublishingPost: select( 'core/editor' ).isPublishingPost(),
-		wasStatus: post.status,
-	};
+	return Promise.resolve( true );
+};
 
-	subscribe( () => {
-		const isPublishingPost = select( 'core/editor' ).isPublishingPost();
-		const isStatus = select( 'core/editor' ).getCurrentPost().status;
+/**
+ * Returns a promise and resolves when the post has been saved and notices have shown.
+ *
+ * @return {Promise} Promise for overlay existence.
+ */
+const saveCompleted = () => {
+	if (
+		null === document.querySelector( '.components-notice__content' ) &&
```

Comment:
One concern I have is another plugin creating a notice that could give a false positive here.  Maybe we could update the selector to `.post-publish-panel__postpublish` which I think is only visible after publishing.